### PR TITLE
Require coalesce to have at least two arguments

### PR DESCRIPTION
**Pull request by electrum**
The SQL specification syntactically allows a single argument, but does
not contemplate it in the description of the semantics. Most commercial
databases require at least two arguments, which indicates that this is
a correct interpretation of the specification.

### CODE REVIEW

**Comment by nezihyigitbasi** [comment]
test failure looks related

File: presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java [hunk 1]
```diff
@@ -408,6 +409,15 @@ public void testArithmeticUnary()
     }
 
     @Test
+    public void testCoalesce()
+    {
+        assertInvalidExpression("coalesce()", "The 'coalesce' function must have at least two arguments");
+        assertInvalidExpression("coalesce(5)", "The 'coalesce' function must have at least two arguments");
+        assertExpression("coalesce(13, 42)", new CoalesceExpression(new LongLiteral("13"), new LongLiteral("42")));
+        assertExpression("coalesce(6, 7, 8)", new CoalesceExpression(new LongLiteral("6"), new LongLiteral("7"), new LongLiteral("8")));
```
Comment (satybald) [inline presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java]: I'd just in case, add a positive test for two null values.
